src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -259,11 +261,31 @@ internal ContractTask<T> CallFromNativeContract<T>(UInt160 callingScriptHash, UI
         protected override void ContextUnloaded(ExecutionContext context)
         {
             base.ContextUnloaded(context);
+            if (context.Script != CurrentContext?.Script)
+            {
+                ExecutionContextState state = context.GetState<ExecutionContextState>();
+                if (UncaughtException is null)
+                {
+                    state.Snapshot?.Commit();
```

Comment:
If the script is the same, where is the Commit?

Reply:
If the script is the same, then it means that it is not a cross-contract exception. In this case, no commit is required since we will not create snapshots for non-cross-contract calls.